### PR TITLE
Feat/marxan 250 cascade deletion of projects to associated resources

### PR DESCRIPTION
**Pull request by angelhigueraacid**
This PR adds the following features:

- `DeleteProject` command. Now when the user tries to delete a project,  a DeleteProject command will be send.
- `ProjectDeleted` event. Once the project has been successfully deleted in `DeleteProjectHandler`, the `ProjectDeleted` event will be emitted.
- `ScheduleCleanupForProjectUnusedResources` command. When the `ProjectDeleted` event is emitted, it triggers the `ScheduleCleanupForProjectUnusedResources` command to be sent.
- `UnusedResourcesCleanupWorker` worker. `ScheduleCleanupForProjectUnusedResourcesHandler` handler will add a job to `UnusedResourcesCleanupWorker` queue. 
- `UnusedResourcesCleanupProcessor` processor. `UnusedResourcesCleanupWorker` worker calls `UnusedResourcesCleanupProcessor` processor to run the jobs submitted to the queue 

### Feature relevant tickets

[[debt] Cascade deletion of projects to all associated resources (project-level geo data, scenarios...)](https://vizzuality.atlassian.net/browse/MARXAN-250)

### CODE REVIEW

File: api/apps/api/src/modules/projects/delete-project/project-deleted.saga.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { ProjectDeleted } from '../events/project-deleted.event';
+import { ScheduleCleanupForProjectUnusedResources } from './schedule-project-unused-resources-cleanup.command';
+
+@Injectable()
+export class ProjectDeletedSaga {
+  @Saga()
+  calculateBlmDefaults = (events$: Observable<any>): Observable<ICommand> =>
```
Comment (hotzevzl) [inline api/apps/api/src/modules/projects/delete-project/project-deleted.saga.ts:12]: probably a copy/paste leftover?
```suggestion
  deleteProjectResources = (events$: Observable<any>): Observable<ICommand> =>
```

**Review by hotzevzl — APPROVED** [review]
looks great, thanks!

Rubén said he's fixing some typos in his current branch which builds on this one, so no need to do anything for these

I'd suggest some minor naming changes:
`DeleteUnusedResources => UnusedResources`, `DeleteProjectUnusedResources => ProjectUnusedResources`, and so on for scenarios, so that in practice we'd then call `removeForProject(projectId, data)` or `removeForScenario(scenarioId)` - these are mostly legibility suggestions, so feel free to decide (or better, if we want to do this at all, it's probably best to just refactor the names in Rubén's PR based on this one.

Basically, I would merge as is :)
